src/page-modules/assistant/trip/index.tsx

```diff
@@ -68,12 +68,22 @@ export default function Trip({ tripQuery, fallback }: TripProps) {
   ): string | undefined => {
     if (!trips) return undefined;
     if (tripIndex === 0 && patternIndex === 0) return undefined;
+
     if (patternIndex === 0) {
-      return trips[tripIndex - 1]?.tripPatterns[
-        trips[tripIndex - 1]?.tripPatterns.length - 1
-      ]?.expectedStartTime;
+      // Iterate backwards to find the first trip with a non-empty tripPatterns list
+      for (let i = tripIndex - 1; i >= 0; i--) {
+        const previousTrip = trips[i];
+        if (previousTrip.tripPatterns.length > 0) {
+          return previousTrip.tripPatterns[previousTrip.tripPatterns.length - 1]
+            ?.expectedStartTime;
+        }
+      }
+      // If no previous trip with non-empty tripPatterns is found
+      return undefined;
```

Comment:
```suggestion
      return;
```

This would be the same, right? IMO it is cleaner, but maybe less explicit.

Reply:
That should be the same, yes. I'm not sure if I agree with the change, but not having a strong opinion either 😄 . What do you think, @jonasbrunvoll ☝️

Reply:
I prefer `returner undefined` over `return`, as I think it is easier to read when the return property is written explicitly.

Reply:
Today in "JavaScript History":
> In pre-ESM module times where strict mode (e.g. `"use strict";`) was not default JavaScript behaviour, `undefined` was not a reserved keyword. Making it so that any external dependency could override the value of undefined with for instance: `undefined = true;` and breaking things. Due to this it was recommended often to use either just `return;` which is implicitly `return undefined;`, or explicitly `return void 0;` as `void` is an operator in JavaScript making every operand return undefined. It could technically also have been `return void 'mikael';` or what ever as the operand, but `void 0` was common practise as it communicated abcense of anything.